lib/plugins/aws/package/compile/functions/index.js

```diff
@@ -210,12 +210,27 @@ class AwsCompileFunctions {
         }
       }
 
+      let kmsKeyArn;
       const serviceObj = this.serverless.service.serviceObject;
-      if (functionObject.awsKmsKeyArn || (serviceObj && serviceObj.awsKmsKeyArn)) {
-        const arn = functionObject.awsKmsKeyArn || (serviceObj && serviceObj.awsKmsKeyArn);
+      if (functionObject.kmsKeyArn || this.serverless.service.provider.kmsKeyArn) {
+        kmsKeyArn = functionObject.kmsKeyArn || this.serverless.service.provider.kmsKeyArn;
+      } else if (functionObject.awsKmsKeyArn) {
+        this.serverless._logDeprecation(
+          'FUNCTION_DEPRECATED_PROPERTY_AWS_KMS_KEY_ARN',
+          'Starting with next major version, awsKmsKeyArn function property will be replaced by kmsKeyArn.'
+        );
```

Comment:
It'll be nice to show this deprecation _unconditionally_ when `functionObject.awsKmsKeyArn` is set

(currently we show it only if it's set and there's no `functionObject.kmsKeyArn` or `this.serverless.service.provider.kmsKeyArn`, which introduces a possibility of throwing config validation error with next major without any deprecation notice in this major)